static/src/javascripts/projects/common/modules/discussion/comment-box.js

```diff
@@ -0,0 +1,653 @@
+
+        if (previewBody && resp && resp.commentBody) {
+            previewBody.innerHTML = resp.commentBody;
+        }
+    }
+
+    /* TODO: remove the replace, get the Scala to be better */
+    getDiscussionId(): string {
+        return (
+            this.options.discussionId ||
+            this.elem
+                .getAttribute('data-discussion-key')
+                .replace('discussion', '')
+        );
+    }
+
+    setFormState(disabled?: boolean): void {
+        const isDisabled = typeof disabled === 'boolean' ? disabled : false;
+
+        const commentBody = this.getElem('body');
+        const submitButton = this.getElem('submit');
+
+        if (isDisabled || commentBody.value.length === 0) {
+            submitButton.setAttribute('disabled', 'disabled');
+        } else {
+            submitButton.removeAttribute('disabled');
+        }
+    }
+
+    setExpanded(): void {
+        this.setState('expanded');
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    getUserData(): Object {
+        return IdentityApi.getUserFromCookie();
+    }
+
+    /* TODO (jamesgorrie); Make this more robust */
+    fail(xhr: Object): void {
+        let response;
+
+        // if our API is down, it returns HTML
+        // this is not so good for JSON.parse
+        try {
+            response = JSON.parse(xhr.responseText);
+        } catch (e) {
+            response = {};
+        }
+
+        this.setFormState();
+
+        if (xhr.status === 0) {
+            this.error('API_CORS_BLOCKED');
+        } else if (response.errorCode === 'EMAIL_NOT_VALIDATED') {
+            this.invalidEmailError();
+        } else if (response.errorCode === 'IP_ADDRESS_BLOCKED') {
+            this.error('IP_THROTTLED');
+        } else if (this.errorMessages[response.errorCode]) {
+            this.error(response.errorCode);
+        } else {
+            this.error('API_ERROR', this.errorMessages.API_ERROR + xhr.status); // templating would be ideal here
+        }
+    }
+
+    clearErrors(): void {
+        this.getElem('messages').innerHTML = '';
+        this.errors = [];
+        this.removeState('invalid');
+    }
+
+    postCommentSuccess(comment: commentType, resp: Object): void {
+        if (this.options.newCommenter) {
+            this.refreshUsernameHtml();
+            this.options.newCommenter = false;
+        }
+
+        comment.id = parseInt(resp.message, 10);
+        this.getElem('body').value = '';
+        this.resetPreviewComment();
+        this.setFormState();
+        this.emit('post:success', comment);
+        mediator.emit('discussion:commentbox:post:success', comment);
+    }
+
+    previewCommentSuccess(comment: commentType, resp: Object): void {
+        this.getElem('preview-body').innerHTML = resp.commentBody;
+        this.setState('preview-visible');
+    }
+
+    /* TODO (jamesgorrie): Perhaps change error states to use bit operators */
```

Comment:
I think all the TODOs in this module will never get done, they are either too vague or don't add enough value. I am in favour of removing (or perhaps, for example, adding to the inbox and given a priority of `0` 😄 )